### PR TITLE
parser: support `ALTER DATABASE t SET TIFLASH REPLICA` in grammar

### PR DESCRIPTION
**Pull request by CalvinNeo**
### What problem does this PR solve?

Issue Number: ref #32254

Problem Summary:
This PR is part of batch setting TiFlash replica in one database. It only supports the grammar, and does nothing.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
support `ALTER DATABASE t SET TIFLASH REPLICA` in grammar
```

### CODE REVIEW

File: parser/ast/ddl.go [hunk 1]
```diff
@@ -73,14 +73,20 @@ const (
...
-	Tp        DatabaseOptionType
-	Value     string
-	UintValue uint64
+	Tp             DatabaseOptionType
+	Value          string
+	UintValue      uint64
+	TiFlashReplica *TiFlashReplicaSpec
+}
+
+func restoreTiFlashStmt(ctx *format.RestoreCtx, tiFlashReplica *TiFlashReplicaSpec) {
```
Comment (xhebox) [inline parser/ast/ddl.go:88]: What is this line?

File: parser/parser.y [hunk 2]
```diff
@@ -3667,6 +3667,17 @@ DatabaseOption:
 			UintValue: placementOptions.UintValue,
 		}
 	}
+|	"SET" "TIFLASH" "REPLICA" LengthNum LocationLabelList
```
Comment (zimulala) [inline parser/parser.y:3670]: If we do it here, we need to support `"SET" "TIFLASH" "REPLICA" LengthNum LocationLabelList "SET" "TIFLASH" "REPLICA" LengthNum LocationLabelList`. 
Is this expected? Add relevant tests if expected.

File: parser/ast/ddl.go (hunk 1 above)
Comment (CalvinNeo) [reply parser/ast/ddl.go:88]: Some later work...I should remove them before have this PR...

**Review by xhebox — APPROVED** [review]

File: parser/parser.y (hunk 2 above)
Comment (zimulala) [reply parser/parser.y:3670]: Could we add some tests to `parser_test.go`?

File: ddl/ddl_api.go [hunk 3]
```diff
@@ -262,14 +272,35 @@ func checkAndNormalizePlacementPolicy(ctx sessionctx.Context, placementPolicyRef
...
+	if len(specs) > 1 {
+		for _, spec := range specs {
+			if spec.Tp == ast.DatabaseSetTiFlashReplica {
+				if hasSetTiFlashReplica {
+					return errRunMultiSchemaChanges
+				}
+				hasSetTiFlashReplica = true
+			}
+		}
+	}
```
Comment (zimulala) [inline ddl/ddl_api.go:286]:
```suggestion
	if len(specs) == 1 {
	    return nil
	}
	for _, spec := range specs {
		if spec.Tp == ast.DatabaseSetTiFlashReplica {
			if hasSetTiFlashReplica {
				return errRunMultiSchemaChanges
			}
			hasSetTiFlashReplica = true
		}
	}
```

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**